pkg/gpg/gpg_forwarding.go

```diff
@@ -0,0 +1,126 @@
+package gpg
+
+import (
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+type GPGConf struct {
+	PublicKey  []byte
+	OwnerTrust []byte
+	SocketPath string
+}
+
+func (g *GPGConf) StopGpgAgent() error {
+	return exec.Command("gpgconf", []string{"--kill", "gpg-agent"}...).Run()
+}
+
+func (g *GPGConf) ImportGpgKey() error {
+	gpgImportCmd := exec.Command("gpg", "--import")
+
+	stdin, err := gpgImportCmd.StdinPipe()
+	if err != nil {
+		return err
+	}
+
+	go func() {
+		defer stdin.Close()
+		_, _ = stdin.Write(g.PublicKey)
+	}()
+
+	return gpgImportCmd.Run()
+}
+
+func (g *GPGConf) ImportOwnerTrust() error {
+	gpgOwnerTrustCmd := exec.Command("gpg", "--import-ownertrust")
+
+	stdin, err := gpgOwnerTrustCmd.StdinPipe()
+	if err != nil {
+		return err
+	}
+
+	go func() {
+		defer stdin.Close()
+		_, _ = stdin.Write(g.OwnerTrust)
+	}()
+
+	return gpgOwnerTrustCmd.Run()
+}
+
+func (g *GPGConf) SetupGpgConf() error {
+	_, err := os.Stat(filepath.Join(os.Getenv("HOME"), ".gnupg", "gpg.conf"))
+	if err != nil {
+		_, err = os.Create(filepath.Join(os.Getenv("HOME"), ".gnupg", "gpg.conf"))
+		if err != nil {
+			return err
+		}
+	}
+
+	gpgConfig, err := os.ReadFile(filepath.Join(os.Getenv("HOME"), ".gnupg", "gpg.conf"))
+	if err != nil {
+		return err
+	}
+
+	if !strings.Contains(string(gpgConfig), "use-agent") {
+		f, err := os.OpenFile(filepath.Join(os.Getenv("HOME"), ".gnupg", "gpg.conf"),
+			os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0o600)
+		if err != nil {
+			return err
+		}
+		defer f.Close()
+
+		if _, err := f.WriteString("use-agent\n"); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (g *GPGConf) SetupRemoteSocketDirtree() error {
+	err := exec.Command("sudo", "mkdir", "-p", "/run/user", filepath.Dir(g.SocketPath)).Run()
+	if err != nil {
+		return err
+	}
+
+	return exec.Command("sudo",
+		"chown",
+		"-R",
+		strconv.Itoa(os.Getuid())+":"+strconv.Itoa(os.Getgid()),
+		"/run/user",
+		filepath.Dir(g.SocketPath),
+		g.SocketPath,
+	).Run()
+}
+
+func (g *GPGConf) SetupRemoteSocketLink() error {
```

Comment:
maybe a better structure or comment what that function is doing? It's not clear at the first glance, what is happening to those symlinks etc :)

Reply:
Done :+1: